### PR TITLE
[A11y] Add ability to add native accessibility children to Xwt access…

### PR DESCRIPTION
**Pull request by iainx**
…ibles

Sometimes widgets need accessible children that aren't backed by individual
widgets, this allows them to be created natively and added to Xwt Widgets

### CODE REVIEW

**Comment by Therzok** [comment]
@iainx we should also report the kind of item that was passed in in the exception, like `nativeAccessible.GetType().ToString()`

File: Xwt.Gtk.Mac/GtkMacAccessibleBackend.cs [hunk 1]
```diff
@@ -83,5 +83,53 @@ internal static INSAccessibility GetNSAccessibilityElement (GLib.Object o)
 				nsa.AccessibilityUrl = new NSUrl (value.AbsoluteUri);
 			}
 		}
+
+		public override void AddChild(object childAccessible)
```
Comment (Therzok) [inline Xwt.Gtk.Mac/GtkMacAccessibleBackend.cs]: all others are named `nativeAccessibel`

Comment (iainx) [reply Xwt.Gtk.Mac/GtkMacAccessibleBackend.cs]: good point

File: Xwt.Gtk.Mac/GtkMacAccessibleBackend.cs [hunk 2]
```diff
@@ -83,5 +83,53 @@ internal static INSAccessibility GetNSAccessibilityElement (GLib.Object o)
...
+			if (nativeObject == null) {
+				throw new ArgumentException ("Not NSObject", nameof (nativeAccessible));
+			}
+
+			var nsa = GetNSAccessibilityElement (widget.Accessible);
+			if (nsa == null) {
+				return;
+			}
+
+			var children = nsa.AccessibilityChildren.ToList ();
```
Comment (sevoku) [inline Xwt.Gtk.Mac/GtkMacAccessibleBackend.cs]: See the comments here: https://github.com/mono/xwt/pull/747/files#r148128485

File: Xwt.XamMac/Xwt.Mac/AccessibleBackend.cs [hunk 3]
```diff
@@ -173,6 +174,37 @@ bool PerformPress ()
...
+
+		public void RemoveChild (object nativeAccessible)
+		{
+			var nativeObject = nativeAccessible as NSObject;
+
+			if (nativeObject == null) {
+				throw new ArgumentException ("Not NSObject", nameof (nativeAccessible));
+			}
+
+			var children = widget.AccessibilityChildren.ToList ();
```
Comment (sevoku) [inline Xwt.XamMac/Xwt.Mac/AccessibleBackend.cs]: also here: https://github.com/mono/xwt/pull/747/files#r148128485

**Comment by iainx** [comment]
Work on #747 instead

**Pull request closed without merging**